### PR TITLE
Feature/ins-1518-improve-create-request-dropdownmodal

### PR DESCRIPTION
**Pull request by jackkav**
Request shortcuts in new request/folder dropdown. It was hard to find gRPC when it was nestled in the http methods dropdown. This aims to raise the discoverability of gRPC, GraphQL and in future websockets.

Changes
- ~~New request modal previously showed mime type selection on post put and patch hiding it for other methods, this isn't suitable for graphql. Therefore I have elected to remove the hiding and showing of the mime type selection in order to simplify the use cases of this form.~~

Friction
- ~~the existing mime type show/hide UX is too twitchy, there is only one reasonable case for it to be disabled/hidden which is grpc, however I would argue this is still the wrong place for GRPC to go since the other options are all HTTP methods. This logic holds true for websockets too. Therefore I have gone with the approach above.~~
- there's a lot of prop drilling and indirection in this code path, related to hotkeys.
- open to suggestions about default hotkeys for grpc and graphql...
- ~~the ticket INS-1518 suggests removing the New Request modal flow entirely but that would conflate insomnia approach to request naming, that could be scoped out of this PR and still deliver on increased discoverability. However removing modal code could be a "good thing ™" since we have issues with the implementation of modals.~~

**New Dropdown**
image

**Old Modal**
image

**Update: removed new request modal** 
In light of this change a few new issues arise.
Friction
- new folder flow is now inconsistent and has different rename flow.
- grpc still has a set proto modal, which could be delayed until the url is filled out.
- quick create request hotkey is now identical to create request, so I've renamed it to create http request.
- should folder context menu also contain these quick request types? **Not yet.**
- should I transform the two dropdown components to function components? would entail untangling the dropdown hide show ref drilling.

changelog(Improvements): Improved the create new request shortcuts dropdown

### CODE REVIEW

File: packages/insomnia/src/common/hotkeys.ts [hunk 1]
```diff
@@ -206,14 +205,10 @@ const defaultRegistry: HotKeyRegistry = {
     keyComb(false, false, false, true, keyboardKeys.k.keyCode),
     keyComb(true, false, false, false, keyboardKeys.k.keyCode),
   ),
-  [hotKeyRefs.REQUEST_SHOW_CREATE.id]: keyBinds(
+  [hotKeyRefs.REQUEST_CREATE_HTTP.id]: keyBinds(
     keyComb(false, false, false, true, keyboardKeys.n.keyCode),
     keyComb(true, false, false, false, keyboardKeys.n.keyCode),
   ),
-  [hotKeyRefs.REQUEST_QUICK_CREATE.id]: keyBinds(
```
Comment (dimitropoulos) [inline packages/insomnia/src/common/hotkeys.ts:213]: Why was this removed?  I use it all the time.  was it moved somewhere else maybe?

Comment (jackkav) [reply packages/insomnia/src/common/hotkeys.ts:213]: Thanks for the early review! Forgot to add that to the PR description. I've renamed them both to point at REQUEST_CREATE_HTTP.

Comment (jackkav) [reply packages/insomnia/src/common/hotkeys.ts:213]: I merged them.

**Review by filfreire — APPROVED** [review]

**Review by dimitropoulos — CHANGES_REQUESTED** [review]
hey: reminder that we want analytics events for stuff like this

**Review by gatzjames — APPROVED** [review]
Seems that most of the friction points have been moved to new issues 👏
Imo migrating the dropdown seems like a big task for this PR so we could keep it out of scope for now.
Tested and it works great! 🚀

**Comment by jackkav** [comment]
Tested analytics from new request in folder, add button and empty state. All send successfully @dimitropoulos

**Review by dimitropoulos — DISMISSED** [review]
I know I know we're trying to get this PR over the line and I hate having to ask this.. but what's the story with the request group menu not having the graphql and grpc items anymore in the last commit?  Apologies if I missed it somewhere.



___

Also, independent of that, the change to put unbound functions in callbacks (e.g. `onClick={() => someThing()}}`) has pretty sever performance problems it presents (over time, especially).

___

I did, however, confirm that the segment events area all working now, so that's good at least.

**Comment by jackkav** [comment]
It wasn't in the ticket, so I thought to add it if it was straightforward. However once I did the number of new decisions to make increased the scope too much for a single PR.

**Review by marckong — APPROVED** [review]

**Pull request merged**